hewr/R/directory_utils.R

```diff
@@ -37,12 +36,29 @@ parse_model_batch_dir_name <- function(model_batch_dir_name) {
     )
   }
 
-  return(list(
+  result <- list(
     disease = disease_map_lower[[matches[2]]],
-    report_date = lubridate::ymd(matches[3]),
-    first_training_date = lubridate::ymd(matches[4]),
-    last_training_date = lubridate::ymd(matches[5])
-  ))
+    report_date = lubridate::ymd(matches[3], quiet = TRUE),
+    first_training_date = lubridate::ymd(matches[4], quiet = TRUE),
+    last_training_date = lubridate::ymd(matches[5], quiet = TRUE)
+  )
+
+  if (any(sapply(result, is.null)) || any(sapply(result, is.na))) {
```

Comment:
```suggestion
  if (any(sapply(result, is.null)) || any(is.na(result))) {
```
`is.na` works as expected with lists.

Reply:
You would need 
```r
disease_map_lower <- c(
  "covid-19" = "COVID-19",
  "influenza" = "Influenza"
)
```
and
```r
parse_model_batch_dir_path <- function(model_batch_dir_path) {
  pattern <- "(.+)_r_(.+)_f_(.+)_t_(.+)"
  model_batch_dir_name <- fs::path_file(model_batch_dir_path)
  matches <- stringr::str_match(
    model_batch_dir_name,
    pattern
  )
  
  if (any(is.na(matches))) {
    stop(
      "Invalid format for model batch directory name; ",
      "could not parse. Expected ",
      "'<disease>_r_<report_date>_f_<first_training_date>_t_",
      "<last_training_date>'."
    )
  }
  
  result <- list(
    disease = disease_map_lower[matches[2]],
    report_date = lubridate::ymd(matches[3], quiet = TRUE),
    first_training_date = lubridate::ymd(matches[4], quiet = TRUE),
    last_training_date = lubridate::ymd(matches[5], quiet = TRUE)
  )
  
  if (any(is.na(result))) {
    stop(
      "Could not parse extracted disease and/or date ",
      "values expected 'disease' to be one of 'covid-19' ",
      "and 'influenza' and all dates to be valid dates in ",
      "YYYY-MM-DD format. Got: ",
      glue::glue(
        "disease: {matches[[2]]}, ",
        "report_date: {matches[[3]]}, ",
        "first_training_date: {matches[[4]]}, ",
        "last_training_date: {matches[[5]]}."
      )
    )
  }
  return(result)
}
```